### PR TITLE
docs: add info on preserveWhitespaces to hydration guide

### PR DESCRIPTION
**Pull request by jessicajaniuk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/angular-compiler-options.md [hunk 1]
```diff
@@ -165,6 +165,12 @@ Do this when using factory summaries.
 When `false`, the default, removes blank text nodes from compiled templates, which results in smaller emitted template factory modules.
 Set to `true` to preserve blank text nodes.
 
+<div class="alert is-helpful">
+
+When using hydration, it is recommended that you use `preserveWhitespaces: false`, which is the default value. If you choose to enable preserving whitespaces by adding `preserveWhitespaces: true` to you tsconfig, it is possible you may encounter issues with hydration. This is not yet a fully supported configuration. Ensure this is also consistently set between the server and client tsconfig files. See the [hydration guide](guide/hydration#preserve-whitespaces) for more details.
```
Comment (AndrewKushnir) [inline aio/content/guide/angular-compiler-options.md:170]:
```suggestion
When using hydration, it is recommended that you use `preserveWhitespaces: false`, which is the default value. If you choose to enable preserving whitespaces by adding `preserveWhitespaces: true` to your tsconfig, it is possible you may encounter issues with hydration. This is not yet a fully supported configuration. Ensure this is also consistently set between the server and client tsconfig files. See the [hydration guide](guide/hydration#preserve-whitespaces) for more details.
```

File: aio/content/guide/hydration.md [hunk 2]
```diff
@@ -91,6 +91,17 @@ There are a few cases where if you have a component template that does not have
 
 Other examples of this behavior are adding a `<div>` inside a `<p>` tag, putting a `<p>` inside of an `<a>`, and putting a `<p>` inside an `<h1>`. If you are uncertain about whether your HTML is valid, you can use a [syntax validator](https://validator.w3.org/) to check it.
 
+<a id="preserve-whitespaces"></a>
+### Preserve Whitespaces Configuration
+
+When using the hydration feature, we recommend using the default setting of `false` for `preserveWhitespaces`. If this setting is not in your tsconfig, the value will be `false` and no changes are required. If you choose to enable preserving whitespaces by adding `preserveWhitespaces: true` to you tsconfig, it is possible you may encounter issues with hydration. This is not yet a fully supported configuration.
```
Comment (AndrewKushnir) [inline aio/content/guide/hydration.md:97]:
```suggestion
When using the hydration feature, we recommend using the default setting of `false` for `preserveWhitespaces`. If this setting is not in your tsconfig, the value will be `false` and no changes are required. If you choose to enable preserving whitespaces by adding `preserveWhitespaces: true` to your tsconfig, it is possible you may encounter issues with hydration. This is not yet a fully supported configuration.
```

**Review by AndrewKushnir — APPROVED** [review]
👍

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 30ec56ec979d60805736f8c864ad4e33e309d777.

**Pull request closed without merging**